### PR TITLE
[CDAP-20202] Add retries in FileLocalizer

### PR DESCRIPTION
**Pull request by rmstar**
File localizer should retry downloading artifacts if appfabric is unavailable.

### CODE REVIEW

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java [hunk 1]
```diff
@@ -40,22 +41,31 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipInputStream;
+import javax.naming.ServiceUnavailableException;
```
Comment (albertshau) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java]: I assume you meant to use the cdap one? Though that is in cdap-common which shouldn't be a dependency here.

You can also catch RetryableException, which is in cdap-api.

Comment (rmstar) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java]: Thanks, fixed.

File: cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java [hunk 2]
```diff
@@ -158,4 +174,28 @@ private Path expand(URI uri, InputStream inputStream, Path targetDir) throws IOE
...
+   * TODO(CDAP-20236): Move Retries code from cdap-common into cdap-retries module and use it here.
+   *
+   * @param callable the callable to run
+   */
+  private void callWithRetries(Callable<Void> callable) throws Exception {
+    int retries = 0;
+    while (true) {
+      try {
+        callable.call();
+        return;
```
Comment (chtyim) [inline cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java:189]: I don't think this is needed.

Comment (chtyim) [reply cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/FileLocalizer.java:189]: Nevermind. It's for breaking the loop.

**Review by albertshau — APPROVED** [review]

**Pull request merged**